File: PalindromeNumber.py


class Solution:
    
    def isPalindrome(self, x):
        xx = x < 0 and -x or x
        tmp_xx = xx
        p_xx = 0

        while xx > 0:
            p_xx = p_xx * 10 + xx % 10
            xx = xx / 10

        return p_xx == tmp_xx


def main():
    s = Solution()
    test = -1234554321
    print s.isPalindrome(test)
    

if __name__ == '__main__':
       main() 

File: src/main/scala/PalindromeNumber/PalindromeNumber.scala
package PalindromeNumber

/**
	* 解题报告
	* 说明： 逆序比较相等就是回文数
	*/

object Solution {

		def reverse(x: Int) = {
			var p = x
			var r = 0
			while(p != 0) {
				r = r * 10 + p % 10 
				p = p / 10
			}
			r
		}

    def isPalindrome(x: Int): Boolean = {
			if (x < 0) false
			else x == reverse(x)
    }
}